homeassistant/components/reisinger_intellidrive/coordinator.py

```diff
@@ -0,0 +1,48 @@
+"""Coordinator for Reisinger Intellidrive."""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+import reisingerdrive
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IntellidriveCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching data from intellidrive device."""
+
+    config_entry: ConfigEntry
+    device: reisingerdrive.ReisingerSlidingDoorDeviceApi
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize ReisingerCoordinator."""
+        self.device = reisingerdrive.ReisingerSlidingDoorDeviceApi(
+            config_entry.data.get(CONF_HOST),
+            config_entry.data.get(CONF_TOKEN),
+            async_get_clientsession(hass),
+        )
+        self.config_entry = config_entry
```

Comment:
```suggestion
    def __init__(self, hass: HomeAssistant, device: ReisingerSlidingDoorDeviceApi) -> None:
        """Initialize ReisingerCoordinator."""
        self.device = device
```

The device should be passed in.